### PR TITLE
FIX: topic owner should watch the new topic when moving posts to a new topic

### PR DESCRIPTION
**Pull request by OsamaSayegh**
https://meta.discourse.org/t/when-a-post-is-moved-to-a-new-topic-by-a-moderator-the-moved-posts-author-is-not-watching/51779/3?u=osama

### CODE REVIEW

File: app/models/post_mover.rb [hunk 1]
```diff
@@ -34,6 +34,8 @@ def to_new_topic(title, category_id = nil, tags = nil)
       )
       DiscourseTagging.tag_topic_by_names(new_topic, Guardian.new(user), tags)
       move_posts_to new_topic
+      force_user_to_watch_new_topic
```
Comment (tgxworld) [inline app/models/post_mover.rb]: This can just be `watch_new_topic` IMO.

File: spec/requests/topics_controller_spec.rb [hunk 2]
```diff
@@ -116,6 +119,30 @@
           expect(Tag.all.pluck(:name)).to contain_exactly("tag1", "tag2")
         end
 
+        it "forces resulting topic owner to watch the new topic" do
```
Comment (tgxworld) [inline spec/requests/topics_controller_spec.rb]: I don't think this should be tested in a request spec. It should instead be tested in `post_mover_spec`.

**Comment by OsamaSayegh** [comment]
@tgxworld done ✅ 

Thanks for your feedback <3

**Comment by tgxworld** [comment]
@OsamaSayegh Looks good :+1:

**Pull request merged**